amethyst_utils/src/fly_camera.rs

```diff
@@ -0,0 +1,107 @@
+use amethyst_core::cgmath::{Deg, Euler, Quaternion, Vector3};
+use amethyst_core::timing::Time;
+use amethyst_core::transform::Transform;
+use amethyst_input::InputHandler;
+use amethyst_renderer::{Camera, ScreenDimensions};
+use specs::{Fetch, Join, ReadStorage, System, WriteStorage};
+
+/// The system that manages the camera movement
+pub struct FlyCameraMovementSystem<'a> {
+    /// The movement speed of the camera in units per second.
+    speed: f32,
+    /// The name of the input axis to locally move in the x coordinates.
+    move_x_name: Option<&'a str>,
+    /// The name of the input axis to locally move in the y coordinates.
+    move_y_name: Option<&'a str>,
+    /// The name of the input axis to locally move in the z coordinates.
+    move_z_name: Option<&'a str>,
+}
+
+impl<'a> FlyCameraMovementSystem<'a> {
+    pub fn new(
+        speed: f32,
+        move_x_name: Option<&'a str>,
+        move_y_name: Option<&'a str>,
+        move_z_name: Option<&'a str>,
+    ) -> Self {
+        FlyCameraMovementSystem {
+            speed,
+            move_x_name,
+            move_y_name,
+            move_z_name,
+        }
+    }
+
+    fn get_axis(name: Option<&'a str>, input: &Fetch<InputHandler<String, String>>) -> f32 {
```

Comment:
don't use specs-stuff in function signatures where not needed, you could take `&InputHandler..>` here. Also, this removes the ability to use a different type for the Axis/Action in InputHandler, we should not force `<String, String>` in our core functionality

Reply:
done